### PR TITLE
tests: added `get_angle` test function in geometry/vector3d

### PR DESCRIPTION
**Pull request by lasgel**
#### Description of Change
a function get_angle is added for the vector3d to calculate the angle between two given vector
test and doc added for this function as well

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C/pull/838"></a>

### CODE REVIEW

File: geometry/vectors_3d.c [hunk 1]
```diff
@@ -223,6 +246,10 @@ static void test()
     assert(fabsf(c.x - (-1.f)) < 0.01);
     assert(fabsf(c.y - (2.f)) < 0.01);
     assert(fabsf(c.z - (-1.f)) < 0.01);
+
+	double alpha = get_angle(&a, &b);
+	//printf("The angle is %f\n", alpha);
```
Comment (Panquesito7) [inline geometry/vectors_3d.c:251]:
```suggestion
	// printf("The angle is %f\n", alpha);
```

File: geometry/vectors_3d.c [hunk 2]
```diff
@@ -191,6 +191,29 @@ mat_3x3 get_cross_matrix(const vec_3d *a)
...
 
+/**
+ * Obtain the angle between two given vectors.
+ * @f[\alpha=acos\left(\frac{\vec{a} \cdot \vec{b}}{\lVert\vec{a}\rVert \cdot \lVert\vec{b}\rVert}\right)@f]
+ * @param[in] a first input vector
+ * @param[in] b second input vector
+ * @returns angle between  @f$\vec{a}@f$ and @f$\vec{b}@f$ in radians
+ */
+
+double get_angle(const vec_3d *a, const vec_3d *b)
```
Comment (Panquesito7) [inline geometry/vectors_3d.c:202]: The indentation in this function (inside) is wrong. Please fix it.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: geometry/vectors_3d.c (hunk 2 above)
Comment (lasgel) [reply geometry/vectors_3d.c:202]: You meant that I used tabs instead of four spaces?
If yes, then the problem was resolved

Comment (Panquesito7) [reply geometry/vectors_3d.c:202]: No, the indentation is wrong, but you must use TABS here instead.

Comment (Panquesito7) [reply geometry/vectors_3d.c:202]: It seems the indentation is fine now. 🙂

File: geometry/vectors_3d.c [hunk 3]
```diff
@@ -191,6 +191,29 @@ mat_3x3 get_cross_matrix(const vec_3d *a)
...
+{
+    double alpha, cos_alpha;
+    float norm_a = vector_norm(a); //calc norm of vec a
+    float norm_b = vector_norm(b); //calc norm of vec b
```
Comment (Panquesito7) [inline geometry/vectors_3d.c:206]: Please provide a better description of these variables (see the example below).

```c++
int64_t data = 0;       ///< The value/key of the node.
```

File: geometry/vectors_3d.c [hunk 4]
```diff
@@ -191,6 +191,29 @@ mat_3x3 get_cross_matrix(const vec_3d *a)
...
+    if (fabsf(norm_a) < EPSILON || fabsf(norm_b) < EPSILON) // detect possible division by 0 - the angle is not defined in this case
```
Comment (Panquesito7) [inline geometry/vectors_3d.c:207]:
```suggestion
    if (fabsf(norm_a) < EPSILON || fabsf(norm_b) < EPSILON) /// detect possible division by 0 - the angle is not defined in this case
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Almost there! 😄

File: geometry/vectors_3d.c (hunk 2 above)
Comment (lasgel) [reply geometry/vectors_3d.c:202]: Thanks a lot for your patience - this is my first pull request on github 🙂

File: geometry/vectors_3d.c [hunk 5]
```diff
@@ -191,6 +191,29 @@ mat_3x3 get_cross_matrix(const vec_3d *a)
...
+    double alpha, cos_alpha;
+    float norm_a = vector_norm(a); ///< The norm of vector a
+    float norm_b = vector_norm(b); ///< The norm of vector b
+    if (fabsf(norm_a) < EPSILON || fabsf(norm_b) < EPSILON) /// detect possible division by 0 - the angle is not defined in this case
+    {  
+        return NAN; 
+    }
+
+    cos_alpha = dot_prod(a, b) / (norm_a * norm_b);
+    alpha = acos(cos_alpha); / /delivers the radian
```
Comment (Panquesito7) [inline geometry/vectors_3d.c:213]:
```suggestion
    alpha = acos(cos_alpha); //delivers the radian
```

File: geometry/vectors_3d.c [hunk 6]
```diff
@@ -191,6 +191,29 @@ mat_3x3 get_cross_matrix(const vec_3d *a)
...
+    double alpha, cos_alpha;
+    float norm_a = vector_norm(a); ///< The norm of vector a
+    float norm_b = vector_norm(b); ///< The norm of vector b
+    if (fabsf(norm_a) < EPSILON || fabsf(norm_b) < EPSILON) /// detect possible division by 0 - the angle is not defined in this case
+    {  
+        return NAN; 
+    }
+
+    cos_alpha = dot_prod(a, b) / (norm_a * norm_b);
+    alpha = acos(cos_alpha); //delivers the radian
```
Comment (Panquesito7) [inline geometry/vectors_3d.c:213]:
```suggestion
    alpha = acos(cos_alpha); // delivers the radian
```

**Review by Panquesito7 — APPROVED** [review]
Great work, @lasgel! Thank you for your dedication and contributions to our community! 👍 🎉

**Review by mishraabhinn — APPROVED** [review]

**Pull request merged**